--- a/arch/arm/boot/dts/allwinner/sun8i-h3-nanopi-neo.dts
+++ b/arch/arm/boot/dts/allwinner/sun8i-h3-nanopi-neo.dts
@@ -65,12 +65,45 @@
 &ohci0 {
 	status = "okay";
 };
+&ehci1 {
+	status = "okay";
+};
+
+&ohci1 {
+	status = "okay";
+};
+
+&ehci2 {
+	status = "okay";
+};
+
+&ohci2 {
+	status = "okay";
+};
 
 &usb_otg {
 	status = "okay";
-	dr_mode = "peripheral";
+	dr_mode = "host";
 };
 
 &usbphy {
 	usb0_id_det-gpios = <&pio 6 12 GPIO_ACTIVE_HIGH>; /* PG12 */
 };
+
+&reg_usb0_vbus {
+	status = "okay";
+	/delete-property/ gpio;
+	regulator-always-on;
+};
+
+&reg_usb1_vbus {
+	status = "okay";
+	/delete-property/ gpio;
+	regulator-always-on;
+};
+
+&reg_usb2_vbus {
+	status = "okay";
+	/delete-property/ gpio;
+	regulator-always-on;
+};
